### PR TITLE
Add getsolutions and value methods for solution disaggregation

### PR DESCRIPTION
**Pull request by laradicp**
The behavior of `JuMP.backend` seems to match https://jump.dev/JuMP.jl/stable/reference/models/#JuMP.backend. I couldn't understand why it works differently for `MOI.submit` yet (it also worked when I used the model attribute `SpsColumnsInfo`). So I'm thinking maybe it's something related to MOI methods.

### CODE REVIEW

File: src/spsinfo.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function MOI.get(dest::MOIU.UniversalFallback, attribute::SpsInfo)
+    return get(dest.modattr, attribute, nothing)
+end
+
+MathOptInterface.Utilities.map_indices(
+    ::MathOptInterface.Utilities.IndexMap, x::Vector{SpInfo}
+) = x
+
+value(::AbstractColumnInfo) = return
```
Comment (guimarqu) [inline src/spsinfo.jl:26]: This method should be a fallback that throws an error.

File: src/spsinfo.jl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+value(info::AbstractColumnInfo, x::JuMP.VariableRef) = value(info, x.index)
+value(::AbstractColumnInfo, idx::MOI.VariableIndex) = return
```
Comment (guimarqu) [inline src/spsinfo.jl:29]: I think it should throw an error.

File: src/spsinfo.jl [hunk 3]
```diff
@@ -0,0 +1,29 @@
+struct SpsInfo <: MOI.AbstractModelAttribute end
+MOI.is_set_by_optimize(::SpsInfo) = true
+
+abstract type AbstractColumnInfo end
+struct SpInfo
```
Comment (guimarqu) [inline src/spsinfo.jl:5]: `SpColsInfo`  sounds better ?

File: src/spsinfo.jl [hunk 4]
```diff
@@ -0,0 +1,29 @@
+struct SpsInfo <: MOI.AbstractModelAttribute end
```
Comment (guimarqu) [inline src/spsinfo.jl:1]: `SpsColsInfo` ?

**Review by guimarqu — COMMENTED** [review]
How do you think you can test the interface in BlockDecomposition ?

**Comment by laradicp** [comment]
> How do you think you can test the interface in BlockDecomposition ?

I added some tests, but the usage isn't the same from Coluna. I had to deactivate `MOI.is_set_by_optimize`.

**Review by guimarqu — APPROVED** [review]

**Pull request merged**